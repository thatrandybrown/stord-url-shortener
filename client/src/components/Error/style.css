.error-section {
  font-size: calc(10px + 2vmin);
}

.error-section > * {
  margin: 7px;
}
