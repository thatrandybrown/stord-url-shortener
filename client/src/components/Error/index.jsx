import "./style.css";

const Error = ({
  heading = "Oops!",
  detail = "Something went wrong!",
  link = true,
}) => {
  return (
    <section className="error-section">
      {heading && <h2 style={{ fontSize: "10rem" }}>{heading}</h2>}
      {detail && <p>{detail}</p>}
      {link && <a href="/">Navigate Home</a>}
    </section>
  );
};

export default Error;
