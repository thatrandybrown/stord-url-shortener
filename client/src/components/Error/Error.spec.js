import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Error from "./";

test("renders the Error Component", () => {
  const { container, getByText } = render(<Error />, { wrapper: MemoryRouter });
  expect(container.firstChild).toMatchSnapshot();
  expect(getByText("Oops!")).toBeInTheDocument();
  expect(getByText("Something went wrong!")).toBeInTheDocument();
});

test("renders the Error Component with Custom Heading Text", () => {
  const {
    container,
    queryByText,
  } = render(<Error heading="An Error Occurred" />, { wrapper: MemoryRouter });
  expect(container.firstChild).toMatchSnapshot();
  expect(queryByText("An Error Occurred")).toBeInTheDocument();
  expect(queryByText("Oops!")).toBeNull();
});

test("renders the Error Component with Custom Detail Text", () => {
  const {
    container,
    queryByText,
  } = render(<Error detail="An Error Occurred" />, { wrapper: MemoryRouter });
  expect(container.firstChild).toMatchSnapshot();
  expect(queryByText("An Error Occurred")).toBeInTheDocument();
  expect(queryByText("Oops!")).toBeInTheDocument();
  expect(queryByText("Something went wrong!")).toBeNull();
});

test("renders the Error component without a Heading", () => {
  const { container } = render(<Error heading={null} />, {
    wrapper: MemoryRouter,
  });
  expect(container.firstChild).toMatchSnapshot();
});

test("renders the Error component without a detail text", () => {
  const { container } = render(<Error detail={null} />, {
    wrapper: MemoryRouter,
  });
  expect(container.firstChild).toMatchSnapshot();
});

test("renders the Error component without a Link", () => {
  const { container } = render(<Error link={false} />, {
    wrapper: MemoryRouter,
  });
  expect(container.firstChild).toMatchSnapshot();
});
