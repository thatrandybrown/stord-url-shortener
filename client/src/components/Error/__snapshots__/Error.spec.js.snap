// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Error Component 1`] = `
<section
  class="error-section"
>
  <h2
    style="font-size: 10rem;"
  >
    Oops!
  </h2>
  <p>
    Something went wrong!
  </p>
  <a
    href="/"
  >
    Navigate Home
  </a>
</section>
`;

exports[`renders the Error Component with Custom Detail Text 1`] = `
<section
  class="error-section"
>
  <h2
    style="font-size: 10rem;"
  >
    Oops!
  </h2>
  <p>
    An Error Occurred
  </p>
  <a
    href="/"
  >
    Navigate Home
  </a>
</section>
`;

exports[`renders the Error Component with Custom Heading Text 1`] = `
<section
  class="error-section"
>
  <h2
    style="font-size: 10rem;"
  >
    An Error Occurred
  </h2>
  <p>
    Something went wrong!
  </p>
  <a
    href="/"
  >
    Navigate Home
  </a>
</section>
`;

exports[`renders the Error component without a Heading 1`] = `
<section
  class="error-section"
>
  <p>
    Something went wrong!
  </p>
  <a
    href="/"
  >
    Navigate Home
  </a>
</section>
`;

exports[`renders the Error component without a Link 1`] = `
<section
  class="error-section"
>
  <h2
    style="font-size: 10rem;"
  >
    Oops!
  </h2>
  <p>
    Something went wrong!
  </p>
</section>
`;

exports[`renders the Error component without a detail text 1`] = `
<section
  class="error-section"
>
  <h2
    style="font-size: 10rem;"
  >
    Oops!
  </h2>
  <a
    href="/"
  >
    Navigate Home
  </a>
</section>
`;
